### DISCUSSION TITLE
Change the range of Delay feedback/amount so maximum value is equivalent to setting it just below the spot where it feedbacks itself into a wall of sound

### DISCUSSION THREAD

**Discussion by soymonitus** [SynthstromAudible/DelugeFirmware#527] [Ideas]
By changing this:
delayWorkingState.delayFeedbackAmount = paramFinalValues[Param::Global::DELAY_FEEDBACK - Param::Global::FIRST];

To this:
delayWorkingState.delayFeedbackAmount = (paramFinalValues[Param::Global::DELAY_FEEDBACK - Param::Global::FIRST] >> 1) - (1 << 24);

We are changing the scale of the delay amount to go (thinking on Midi CC values) instead of the range 0-127, to range 0-63. Avoiding then the range 64-127 where it auto-feeds itself and can damage your speakers / ruin your live set.

The problem with this implementation is that if the community feature is toggled, all previous songs will sound different with the new setting, and will behave with less feedback.
And viceversa, a song saved with this new community setting may have a near-maximum delay amount which doesn't feedback and when loaded in a your friend's Deluge which may have this option disabeld, it will blow your friend's ears.

Is this a feature that will make sense for some users?, so maybe some users marry with it set to ON and never look back? Or should I try to implement it so it doesn't allow the user to go past the middle golden knob point (and also ignore learned midi CC about value 63), and also taking care of synth patching in the mod matrix.... oufff seems like a lot of work already hehehe

**Comment by soymonitus** [comment]
Other option is to limit keep the value at 63 no matter how much you increase it from 63 to 127. That is, the effective range is half of the range, but from the middle to maximum there is no audible difference. But that sounds also hacky anyway. I don't know.
The thing is that self-feedback ranges of the delay are super dangerous. Several times I had to turn off the deluge cause I changed som row's delay while performing in Song mode and then I couldn't remember which one I set too high and had to turn off before blowing my ears.

**Reply by soymonitus** [reply]
Maybe save an option to the song's XML and be able to change it for the song you are working on in Settings. And once you save it, it is saved to songs xml

**Comment by m-m-adams** [comment]
Self oscillating delay is a useful musical effect. I think the appropriate solution is to place an exponential curve on the knob so that most of the range is usable and the self oscillation is squeezed into the upper range. Then old songs can be imported fine

**Reply by manysounds** [reply]
Yes please

**Comment by ok-reza** [comment]
keep in mind the self oscillating when using analog delay is way more musical and can be cranked past halfway (even to 50), sounds more pleasant and doesn't overload the output.

the digital delay at or past 50% is problematic. perhaps it needs to have a limiter?

**Comment by duncanrmi** [comment]
the deeper problem here, for me, is that the delay-send & delay-feedback are combined into one delay-amount setting, & while this might have worked for scratch perry & king tubby with their revoxes, it's not quite subtle enough for me.
can these parameters be split? put the feedback under a pad (amount) & have the lower gold knob be the send? or vice-versa?

**Reply by ok-reza** [reply]
i think this would warrant a separate discussion. a lot of my live performance stuff utilizes having delay amount and feedback tied to that one knob.
